src/views/Cart.jsx

```diff
@@ -0,0 +1,201 @@
+
+export default function Cart() {
+  const dispatch = useDispatch();
+  const { existingItems, cartTotal } = useSelector(
+    (state) => state.getCart.cart
+  );
+  const [currentCart, setCurrentCart] = useState({});
+  const [isLoading, setLoading] = useState(false);
+  const clearingCart = useSelector((state) => state.clearCart.loading);
+  const deletingItemCart = useSelector((state) => state.deleteItemCart.loading);
+  const updatingItemCart = useSelector((state) => state.update.loading);
+  const [cartCleared, setCartCleared] = useState(false);
+  const [cartUpdated, setCartUpdated] = useState(false);
+
+  useEffect(() => {
+    dispatch(getCart());
+    setCurrentCart(existingItems);
+  }, [dispatch, isLoading]);
+
+  useEffect(() => {
+    if (cartCleared) {
+      setCartCleared(false);
+      dispatch(getCart());
+    }
+  }, [dispatch, cartCleared]);
+
+  useEffect(() => {
+    if (cartUpdated) {
+      setCartUpdated(false);
+      dispatch(getCart());
+    }
+  }, [dispatch, cartUpdated]);
+
+  let items = {};
+  if (existingItems !== undefined) {
+    items = existingItems;
+  }
+
+  const handleClearCart = () => {
+    dispatch(clearCart())
+      .then(() => {
+        setCartCleared(true);
+      })
+      .catch((error) => {
+        toast.error('Failed to clear cart', error);
+      });
+  };
+
+  const handleDeleteItemCart = async (id) => {
+    setLoading(true);
+    try {
+      await dispatch(deleteItemCart(id));
+      if (Object.values(existingItems).length === 1) {
+        await dispatch(getCart());
+      }
+    } catch (error) {
+      toast.error('Error while removing item from cart:', error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const handleIncrement = (id) => {
+    dispatch(incrementCart(id)).then(() => {
+      setCartUpdated(true);
+    });
+  };
+
+  const handleDecrement = (id, quantity) => {
+    if (quantity > 1) {
+      dispatch(decrementCart(id)).then(() => {
+        setCartUpdated(true);
+      });
+    } else {
+      toast.warn('Quantity cannot be less than 1');
+    }
+  };
+
+  return (
+    <div className="cart-container" data-testid="cart-container">
+      <div className="cart-header">
+        <div className="cart-title">
+          <p className="cart-name">Cart</p>
+        </div>
+      </div>
+      <div className="cart-products-container">
+        {cartTotal === 0 ? (
+          <div className="empty-cart-container">
+            <div className="cart-image">
+              <img src={cartImage} alt="cart" className="cart-icon-cart" />
+            </div>
+            <div className="cart-message-container">
+              <p className="cart-message">Cart is Empty</p>
+            </div>
+          </div>
+        ) : (
+          <div className="cart-filled-container" style={{ display: 'block' }}>
+            <div className="cart-cards-container" data-testid="cart-card">
+              {Object.values(items).map((item) => (
```

Comment:
`Object.values(items)` can you do this in action because this is not best practice? better to return value only to map it or manipulate as the way you but with formatted well